srml/support/src/event.rs

```diff
@@ -282,11 +282,13 @@ macro_rules! __decl_generic_event {
 		#[cfg_attr(feature = "std", derive(Debug))]
 		$(#[$attr])*
 		pub enum RawEvent<$( $generic_param ),* $(, $instance)? > {
-			#[doc(hidden)]
-			$(PhantomData($crate::rstd::marker::PhantomData<$instance>),)?
 			$(
 				$events
 			)*
+			$(
+				#[doc(hidden)]
+				PhantomData($crate::rstd::marker::PhantomData<$instance>),
```

Comment:
Yeah okay, I would either accept a custom implementation of `Encode`/`Decode`. Or you update `parity-codec-derive` and add `codec(skip)` as serde it is doing.

Reply:
* adding codec(skip) to variants is easy, it will involve a bit more work for variants and fields but ok
* implementing Encode/Decode seems hard as first glance because we need to have index.
  hmm or we can add dependency to crates like enum-iterator-derive

Reply:
For `2.`, we do this already in `decl_module` for the `Call` enum. But I much favor option 1.

Reply:
OK I'll implement skip in codec, but would be in favor merging it as it and add a TODO comment to set skip when done

Reply:
Okay. But please work on this directly.